### PR TITLE
Add `region_info` module

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

This is to add `region_info` module. Addressing the need for users to access information about a specific region.

## ✔️ How to Test
```bash
make TEST_ARGS="-vvv region_info" test
```

```yaml
---
- name: Create a Linode instance
  hosts: localhost
  gather_facts: no
  tasks:
    - name: Get Info of a Linode Region
      linode.cloud.region_info:
        id: us-mia
```

### CODE REVIEW

**Review by lgarber-akamai — APPROVED** [review]
Works well in my local testing, nice work!

File: plugins/module_utils/linode_common.py [hunk 1]
```diff
@@ -107,6 +107,7 @@
 class LinodeModuleBase:
     """A base for all Linode resource modules."""
 
+    ## pylint: disable=too-many-positional-arguments
```
Comment (yec-akamai) [inline plugins/module_utils/linode_common.py:110]: It seems we've handled disabling this linter in different places/style: https://github.com/search?q=repo%3Alinode%2Fansible_linode%20too-many-positional-arguments&type=code Could you pull the latest change from dev to have them organized?

**Review by yec-akamai — COMMENTED** [review]
Tests passed locally! Just a small comment:

**Review by jriddle-linode — APPROVED** [review]
Works locally as expected LGTM.

**Pull request merged**